### ISSUE TITLE
Issues with displaying Dwains Dashboard in 2021.12 betas

### ISSUE THREAD

**Issue by thewazz1** [dwainscheeren/dwains-lovelace-dashboard#340]
With each beta of 2021.12 (through beta 3) I have been having issues with Dwains Dashboard displaying at all.  This occurs in the HA iOS app, Safari on MacOS and Chrome on MacOS.  When I clear the browser caches and refresh the page it will show the dashboard correctly, however once I reboot, or refresh the page the dashboard pane will show up completely blank, except for the navigation icons for the dashboard (such as lights, cameras and more-pages, but they all show up blank as well).  The HA sidebar does still show up, and I can go to the default Lovelace page and it shows up correctly.  I looked in the HA logs and this is the only thing that I see that might be applicable:

2021-12-08 15:18:46 ERROR (MainThread) [frontend.js.latest.202112060] http://192.168.86.84:8123/dwains-dashboard/home:0:0 Uncaught

However, this does not show up every time that I refresh, so may not correlate.

**Comment by thewazz1** [comment]
I also looked in the developer tools in Safari, and it is not throwing any errors other than this:

[Error] NotSupportedError: Cannot define multiple custom elements with the same tag name
	define (button-card.js:1:38792)
	Module Code (button-card.js:1:38792)
	evaluate
	moduleEvaluation

But this has been in the logs for quite a while now.  And this other error:

@babel/polyfill is loaded more than once on this page. This is probably not desirable/intended and may have consequences if different versions of the polyfills are applied sequentially. If you do need to load the polyfill more than once, use @babel/polyfill/noConflict instead to bypass the warning. 

That error was on light-entity-card.js

**Comment by dwainscheeren** [comment]
Yeah there are some issues with 2021.12. Need to look how to fix it and when to fix it

**Comment by mmornati** [comment]
Did the same. No error in the console and not actually able to understand why it is not showing anything at all. 
I checked the hacks plugins updates, but nothing interesting foud :S

**Comment by raz0rf0x** [comment]
12.0 launched today and now it's happening.
Suggest holding back from the latest update if you are using DD.

**Comment by mmornati** [comment]
I noticed some errors which are showing in the browser console only the first time we are opening the dwaing dashboard:
```
ha-config-device-page.ts:62 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'name')
    at /frontend_latest/5a012df7.js:593
    at Array.map (<anonymous>)
    at HTMLElement.<anonymous> (/frontend_latest/5a012df7.js:593)
    at HTMLElement._devicesAndFilterDomains (/frontend_latest/app.b6601425.js:850)
    at HTMLElement.value (/frontend_latest/5a012df7.js:613)
    at HTMLElement.update (/frontend_latest/app.b6601425.js:1243)
    at HTMLElement.performUpdate (/frontend_latest/app.b6601425.js:1243)
    at HTMLElement.scheduleUpdate (/frontend_latest/app.b6601425.js:1243)
    at HTMLElement._$EC (/frontend_latest/app.b6601425.js:1243)
```
This one seems not related, it could be something on my side.

```
dwains-dashboard.js:90 Uncaught TypeError: Cannot read properties of undefined (reading 'popup_cards')
    at j.popup_card (/dwains_dashboard/js/dwains-dashboard.js:90)
    at HTMLElement._showDetails (/hacsfiles/content-card-linky/content-card-linky.js?hacstag=3130384471108:224)
    at HTMLElement.<anonymous> (/hacsfiles/content-card-linky/content-card-linky.js?hacstag=3130384471108:86)
```

```
Uncaught TypeError: Cannot read properties of undefined (reading 'popup_cards')
    at j.popup_card (/dwains_dashboard/js/dwains-dashboard.js:90)
    at i (/frontend_latest/app.b6601425.js:850)
    at u (/frontend_latest/a0e8358d.js:1)
    at HTMLElement.value (/frontend_latest/a0e8358d.js:56)
    at M.handleEvent (/frontend_latest/app.b6601425.js:1243)
    at i (/frontend_latest/app.b6601425.js:850)
    at HTMLDivElement.<anonymous> (/frontend_latest/a0e8358d.js:1)
```

I guess there is only a compatibility problem in Hacs component with the latest lovelace

**Comment by WMunschau** [comment]
Same issue after update. Is there already an approximate date for a solution?

**Comment by mmornati** [comment]
> Same issue after update. Is there already an approximate date for a solution?

On my side, I tested a little bit today but not able to know exactly what is the component is not working well. Without logs or errors is hard.
I modified the dashboard javascript to add some information about what is happening, but no idea at the moment what is the root cause... hopefully @dwainscheeren has any idea :)

**Comment by dwainscheeren** [comment]
I'm patching it as we speak for the last 2 days. I think I pathed it but I need to make an official release. For the people who want to try download the 2.0 zip file: https://github.com/dwainscheeren/dwains-lovelace-dashboard/archive/refs/heads/2.0.zip Put that file on top of existing dwains_dashboard folder. Note if you are loading dd from hacs you need to remove dd first in hacs for this alpha test. Then reboot HA and a cache reset. Then it must work with 2021.12

**Comment by dwainscheeren** [comment]
Hope to release the official dd version for this fix in following few days

**Comment by mmornati** [comment]
Thanks, @dwainscheeren that's great, going to test in a minute.
Just as information I just noticed, after a local cache clean, the dashboard was correctly displayed the first time I entered and everything was working... until a page force refresh.

![image](https://user-images.githubusercontent.com/139323/145728850-49106361-fc18-47a3-822f-ea6cdfa8ee16.png)

**Comment by mmornati** [comment]
> I'm patching it as we speak for the last 2 days. I think I pathed it but I need to make an official release. For the people who want to try download the 2.0 zip file: https://github.com/dwainscheeren/dwains-lovelace-dashboard/archive/refs/heads/2.0.zip Put that file on top of existing dwains_dashboard folder. Note if you are loading dd from hacs you need to remove dd first in hacs for this alpha test. Then reboot HA and a cache reset. Then it must work with 2021.12

Yeah is working. I replaced the whole `custom_components/dwains_dashboard`.

What was the problem ?

**Comment by Nag94** [comment]
Thanks for your quick fix ! On my side, it still have the problem, it is really strange, emptying cache, force reload, restart HA, sometime dashboard, sometimes not. I will continue my tests and give to you debug information if revelant.

**Comment by mmornati** [comment]
> Thanks for your quick fix ! On my side, it still have the problem, it is really strange, emptying cache, force reload, restart HA, sometime dashboard, sometimes not. I will continue my tests and give to you debug information if revelant.

On my side, I just needed to clean the cache (both on android and laptop). Do you have any errors in the browser console? Maybe there is anything else that is blocking the dashboard load event sometimes.

**Comment by Nag94** [comment]
> Thanks for your quick fix ! On my side, it still have the problem, it is really strange, emptying cache, force reload, restart HA, sometime dashboard, sometimes not. I will continue my tests and give to you debug information if revelant.

It seems to be ok under chrome, maybe it is just me, but I think the pagers are loading slowly. I will continue the tests on this to be sure this is not my config and the new HA version

**Comment by DerSchdief** [comment]
I have noticed that with version 2.0.5, some add-ons are no longer displayed correctly.
I have the updates, find_my and home_assistant_view Addons and no one is displayed correctly with the version 2.0.5.

![Screenshot_1](https://user-images.githubusercontent.com/69635789/145737488-1c2b26c1-18a7-45a9-b113-980fd7e29bf7.png)
![Screenshot_2](https://user-images.githubusercontent.com/69635789/145737490-fd029989-0e47-476b-8001-d1f84100c0c2.png)

**Comment by Nag94** [comment]
I can confirm, same for me, it looks like there is only one column for displaying some addons.
<img width="1104" alt="Capture d’écran 2021-12-13 à 02 32 19" src="https://user-images.githubusercontent.com/30177389/145739115-616cc147-360b-4f78-ad5c-7a717101ec1c.png">

In this example, the graph take all the length normally.

**Comment by dwainscheeren** [comment]
@Nag94 can you share the page.yaml of that addon? then I can take a look at it.

**Comment by mmornati** [comment]
I don't have any of this single-column problems. Can it be related in the way you are using the template or some custom display options?
![image](https://user-images.githubusercontent.com/139323/145880598-a5c29e78-57a5-48f2-a41c-b717a61fb07b.png)

**Comment by dwainscheeren** [comment]
https://github.com/dwainscheeren/dwains-lovelace-dashboard/releases/tag/v2.0.5 try this version please

**Comment by lurandly** [comment]
I had the addon issue in room addons and more_page_addons. Looks like col-lg defaulted to col-lg-9 on those pages. Adding 

`items_classes: 'col-xs-12 col-sm-12 col-md-12 col-lg-12' `

below 

`- type: custom:dwains-flexbox-card `

in rooms/room/addon.yaml and 04.more_page_addons.yaml fixed it for me.

**Comment by dwainscheeren** [comment]
Thanks, you are right. Will add that code :)

**Comment by AalianKhan** [comment]
Hello,
I am getting an error when I updated to the latest version. 
<img width="424" alt="image" src="https://user-images.githubusercontent.com/39784853/146324486-13a4978c-649b-44b5-9d52-ce52e6e8fdc0.png">
When I checked the cards in dwains-dashboard in custom components I didn't see any folder named dwains-dashboard. 
<img width="692" alt="image" src="https://user-images.githubusercontent.com/39784853/146324675-cfb3976d-9147-4c83-9e5f-f5cf027486cb.png">
Is there a known fix for this yet?
Thank you and I apologize if this is not the correct place to post this

**Comment by SirUli** [comment]
> I am getting an error when I updated to the latest version. 

You are not the only one. I had to restart HA and cleared all caches of the clients, then it worked. Not sure which part fixed it, but I would start with clearing the caches.

**Comment by dwainscheeren** [comment]
Yeah as I mentioned thousands times earlier, please clear your cache after an update its really important to load the new js files. HA caches files like hell for developers.

**Comment by WolfsWelpe** [comment]
Anyone else got the problem that every more_page addon is kinda off and too large or not really wrking anymore in combination with state-switch card since the latest update?

**Comment by dwainscheeren** [comment]
> I had the addon issue in room addons and more_page_addons. Looks like col-lg defaulted to col-lg-9 on those pages. Adding
> 
> `items_classes: 'col-xs-12 col-sm-12 col-md-12 col-lg-12' `
> 
> below
> 
> `- type: custom:dwains-flexbox-card `
> 
> in rooms/room/addon.yaml and 04.more_page_addons.yaml fixed it for me.

**Comment by WolfsWelpe** [comment]
Sorry I totally overlooked that one :(

Thanks mate <3

**Comment by t0bse** [comment]
![image](https://user-images.githubusercontent.com/45834518/146450866-f664de1f-ae0c-43f7-8c2f-4db0c53a3824.png)

new color for the degrees since 2.0.5? they where white before that update

**Comment by dwainscheeren** [comment]
Then you always got the bug active. They where blue from version 0.1 😛

**Comment by Nag94** [comment]
> @Nag94 can you share the page.yaml of that addon? then I can take a look at it.

@dwainscheeren sorry for my late answer, I have added the modification from @lurandly and everything is ok now.

Thanks for your answer and your amazing job ! :-)

**Comment by hpommier** [comment]
Hi @dwainscheeren ! First of all, thanks for your awesome project !
I have the same bug (not resolved with 2.0.5) about missing icons in all dashboards.

![image](https://user-images.githubusercontent.com/49273092/148516818-b88617a9-1d91-488a-9920-3f7a4b49b184.png)

I think I have found a way to resolve the issue by switching the theme used in HA. When I select "backend-selected" instead of "Dwains Light Theme" :

![image](https://user-images.githubusercontent.com/49273092/148517114-394b54f4-f8d7-4742-8f5e-2717d2367cd7.png)

I hope it will be usefull.

**Comment by cpuks** [comment]
Looks like this bug came back in latest 2022.2 frontend - I just got access to few rooms and rest gives me:
`2022-02-06 20:08:44 ERROR (MainThread) [frontend.js.latest.202202030] https://ha.sucha.dev/dwains-dashboard/home?homescreen=1:0:0 Uncaught`

**Comment by M4rt1nCh** [comment]
I can also confirm that 2022.2.7 throws: NotSupportedError: Cannot define multiple custom elements with the same tag name

**Comment by dwainscheeren** [comment]
button card?

**Comment by MAC-R0ch3** [comment]
> button card?

yes

**Comment by M4rt1nCh** [comment]
> button card?

Yes

**Comment by clipse2004** [comment]
I had this problem with several cards.
The solution was simple, comment out.
The dashboard works as usual and the error is gone.

**Comment by M4rt1nCh** [comment]
Just checked again - the issue is not just related to the button card, but also the light-entity card on my side.... :-(

http://HOST:PORT/hacsfiles/light-entity-card/light-entity-card.js?hacstag=168744428460:301:6046 NotSupportedError: Cannot define multiple custom elements with the same tag name
http://HOST:PORT/hacsfiles/button-card/button-card.js?hacstag=146194325342:1:38791 NotSupportedError: Cannot define multiple custom elements with the same tag name
